specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+        <com:Class Id="{GUIDEntryPoint}" DisplayName="Reunion Push" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+## UWP InProc activator
+The UWP inproc activator flow differs from the Win32 inproc activator flow since there is technically no WinMain activation associated with UWPs. Instead the App will have to  override and implement OnBackgroundActivated which is tightly coupled to Windows.UI.Xaml workstreams.
+
+# API Details
+```c# (but really MIDL3)
+    namespace Microsoft.ProjectReunion
+    {
+        // The legacy ActivationKind does not support operations like Push. So we need to define an extension set
+        enum ActivationKindExtension
+        {
+            Launch = 0,
+            Search = 1,
+            ShareTarget = 2,
+            File = 3,
+            Protocol = 4,
+            FileOpenPicker = 5,
+            FileSavePicker = 6,
+            CachedFileUpdater = 7,
+            ContactPicker = 8,
+            Device = 9,
+            PrintTaskSettings = 10,
+            CameraSettings = 11,
+            RestrictedLaunch = 12,
+            AppointmentsProvider = 13,
+            Contact = 14,
+            LockScreenCall = 15,
+            VoiceCommand = 16,
+            LockScreen = 17,
+            PickerReturned = 1000,
+            WalletAction = 1001,
+            PickFileContinuation = 1002,
+            PickSaveFileContinuation = 1003,
+            PickFolderContinuation = 1004,
+            WebAuthenticationBrokerContinuation = 1005,
+            WebAccountProvider = 1006,
+            ComponentUI = 1007,
+            ProtocolForResults = 1009,
+            ToastNotification = 1010,
+            Print3DWorkflow = 1011,
+            DialReceiver = 1012,
+            DevicePairing = 1013,
+            UserDataAccountsProvider = 1014,
+            FilePickerExperience = 1015,
+            LockScreenComponent = 1016,
+            ContactPanel = 1017,
+            PrintWorkflowForegroundTask = 1018,
+            GameUIProvider = 1019,
+            StartupTask = 1020,
+            CommandLineLaunch = 1021,
+            BarcodeScannerProvider = 1022,
+            Push = 1023,
+        };
+
+        // Simply a wrapper around IActivatedEventArgs to support new Kind extensions like Push which aren't a part of Foundations today
+        runtimeclass ActivatedEventArgsExtension
+        {
+            ActivatedEventArgsExtension(Windows.ApplicationModel.Activation.IActivatedEventArgs activatedEventArgs, ActivationKindExtension kind2);
+            ActivationKindExtension Kind2 { get; };
+            Windows.ApplicationModel.Activation.IActivatedEventArgs ActivatedArgs{ get; };
+        }
+
+        // This is an overload of the existing GetActivatedEventArgs in Reunion which supports new Activation Kinds
+        static runtimeclass AppLifecycle2
```

Comment:
Similarly here - remove these types and focus on the stuff this API is going to do; The other content needs to go into @andreww-msft and @aeloros 's docs.